src/main/java/org/folio/config/ApplicationConfig.java

```diff
@@ -35,16 +36,23 @@
 import org.folio.services.piece.PieceClaimingService;
 import org.folio.services.piece.PieceService;
 import org.folio.services.title.TitleService;
+import org.folio.services.user.NoOpUserService;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 @Configuration
 @ComponentScan({ "org.folio.verticles", "org.folio.event.listener" })
-@Import({ KafkaConfiguration.class })
+@Import({ KafkaConfiguration.class, org.folio.spring.ApplicationConfig.class })
 public class ApplicationConfig {
 
+  // Override UserService from folio-custom-fields
```

Comment:
I think we need some improvement here, for example if we would like to integrate custom-fields to invoices - we also need to implement a service to interact with users that is not desired.
Did you try to use Spring Conditional annotation to exclude unnecessary beans from spring context?
Or you can use ConditionalOnProperty annotation, so if custom-fields are used only in mod-users - need to add property only to mod-users and by default in all other places this service will not be included into context.

Reply:
Yes. This is a workaround that can/should be removed once the overlaying issue is resolved. I think the work needs to be done in `folio-custom-fields` itself as it currently is depending on a implementation of `UserService`. But it is not on me to make such a decision. I have filed [FCFIELDS-45](https://issues.folio.org/browse/FCFIELDS-45) for this.
Its more or less the same with the `CustomFieldsPostDeployVerticle` and [FCFIELDS-42](https://issues.folio.org/browse/FCFIELDS-42).

For now, these are the only options I found that allow us to move forward with [MODORDERS-964](https://issues.folio.org/browse/MODORDERS-964).

Reply:
Okay got it, reaching Volaris team, maybe they already have some plans to prioritize these 2 bugs.

Reply:
Thanks!